File: chap17_renal/current_plot.py

#     -------------------------------------------------------------------
# 
#      Co-current/countercurrent comparison.
# 
#      For Chapter 17, Section 17.2 of
#      Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
# 
#      Written by James Keener and James Sneyd.
# 
#     -------------------------------------------------------------------

import numpy as np
import matplotlib.pyplot as plt

alpha = 10
x = np.linspace(0,1,100)

Cco = (1-np.exp(-2*alpha*x))/2
Ccount = alpha*x/(1+alpha)

plt.figure()
plt.plot(x,Cco,x,Ccount)
plt.xlabel('x/L')
plt.ylabel('C_1/C')
plt.text(0.2, 0.55, 'cocurrent', fontsize=12)
plt.text(0.55, 0.8, 'countercurrent', fontsize=12)

File: chap17_renal/current_plot.m

%    -------------------------------------------------------------------
%
%     Co-current/countercurrent comparison.
%
%     For Chapter 17, Section 17.2 of
%     Keener and Sneyd, Mathematical Physiology, 3rd Edition, Springer.
%
%     Written by James Keener and James Sneyd.
%
%    -------------------------------------------------------------------

clear all; close all; clc;
set(0,                           ...
   'defaultaxesfontsize', 20,   ...
   'defaultaxeslinewidth', 1.2, ...
   'defaultlinelinewidth', 2.0); 


alpha = 10;
x = [0:.01:1];

Cco = (1-exp(-2*alpha*x))/2;
Ccount = alpha*x/(1+alpha);

figure(1)
plot(x,Cco,x,Ccount)
xlabel('x/L')
ylabel('C_1/C')
text(.2,.55,'cocurrent','fontsize',18)
text(.55,.8,'countercurrent','fontsize',18)

